lrlex/src/lib/parser.rs

```diff
@@ -121,6 +144,26 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
         }
     }
 
+    fn validate_start_state<'a>(
+        &self,
+        off: usize,
+        name: &'a str,
+    ) -> LexInternalBuildResult<&'a str> {
+        self.validate_start_state_name(off, name)?;
+        if self.start_states.iter().any(|state| state.name == name) {
+            Err(self.mk_error(LexErrorKind::DuplicateStartState, off))
+        } else {
+            Ok(name)
+        }
+    }
+
+    fn validate_start_state_name(&self, off: usize, name: &str) -> LexInternalBuildResult<()> {
+        match RE_START_STATE_NAME.is_match(name) {
```

Comment:
I think this would be clearer as just `if RE_START_STATE_NAME.is_match(name)` (rather than a `match`).

Reply:
I agree - will modify to an if + else

Reply:
Updated in https://github.com/softdevteam/grmtools/pull/318/commits/6ace9f373b657ebe18660cf0c5d0bd2010cc7ba9